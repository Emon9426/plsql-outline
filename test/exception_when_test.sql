CREATE OR REPLACE FUNCTION test_multiple_when RETURN NUMBER IS
BEGIN
    -- 一些代码
    RETURN 1;
EXCEPTION
    WHEN NO_DATA_FOUND THEN
        RETURN 0;
    WHEN TOO_MANY_ROWS THEN
        RETURN -1;
    WHEN VALUE_ERROR THEN
        RETURN -2;
    WHEN OTHERS THEN
        RETURN -999;
END;

CREATE OR REPLACE PROCEDURE complex_exception_proc IS
    v_result NUMBER;
BEGIN
    -- 主要逻辑
    v_result := test_multiple_when();
    
    BEGIN
        -- 嵌套块
        NULL;
    EXCEPTION
        WHEN INVALID_NUMBER THEN
            NULL;
        WHEN OTHERS THEN
            NULL;
    END;
    
EXCEPTION
    WHEN NO_DATA_FOUND THEN
        DBMS_OUTPUT.PUT_LINE('No data found');
    WHEN TOO_MANY_ROWS THEN
        DBMS_OUTPUT.PUT_LINE('Too many rows');
    WHEN OTHERS THEN
        DBMS_OUTPUT.PUT_LINE('Other error');
END;
