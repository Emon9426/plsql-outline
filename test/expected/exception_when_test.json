{
  "nodes": [
    {
      "type": "FUNCTION",
      "name": "test_multiple_when",
      "declarationLine": 1,
      "beginLine": 2,
      "exceptionLine": 5,
      "endLine": 14,
      "level": 1,
      "children": []
    },
    {
      "type": "PROCEDURE",
      "name": "complex_exception_proc",
      "declarationLine": 16,
      "beginLine": 18,
      "exceptionLine": 32,
      "endLine": 39,
      "level": 1,
      "children": []
    }
  ],
  "metadata": {
    "sourceFile": "exception_when_test.sql",
    "totalLines": 33,
    "maxNestingDepth": 1
  }
}